libs/server-sdk/src/data_components/serialization_adapters/json_deserializer.cpp

```diff
@@ -36,7 +36,11 @@ JsonDeserializer::AllSegments() const {
 }
 
 std::string const& JsonDeserializer::Identity() const {
-    return reader_.Identity();
+    return reader_->Identity();
+}
+
+bool JsonDeserializer::Initialized() const {
+    return reader_->Initialized();
 }
```

Comment:
Since `JsonDeserializer` now mirrors all the functions of `ISerializedDataReader` (`Initialized` was missing), we can simply have it own the `shared_ptr<ISerializedDataReader>` instead of taking a reference.

This way the Lazy Load system only needs to own one reader instead of the two (to get at `Initialized`).